viamonvifdiscovery/service.go

```diff
@@ -0,0 +1,138 @@
+// Package viamonvifdiscovery provides the discovery service that wraps ONVIF integration for the viamrtsp module
+package viamonvifdiscovery
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/viam-modules/viamrtsp"
+	"github.com/viam-modules/viamrtsp/viamonvif"
+	"github.com/viam-modules/viamrtsp/viamonvif/device"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/discovery"
+)
+
+// Model is the model for a rtsp discovery service.
+var (
+	Model             = viamrtsp.Family.WithModel("onvif")
+	errNoCamerasFound = errors.New("no cameras found, ensure cameras are working or check credentials")
+)
+
+func init() {
+	resource.RegisterService(
+		discovery.API,
+		Model,
+		resource.Registration[discovery.Service, *Config]{
+			Constructor: newDiscovery,
+		})
+}
+
+// Config is the config for the discovery service.
+type Config struct {
+	Credentials []device.Credentials `json:"credentials"`
+}
+
+// Validate validates the discovery service.
+func (cfg *Config) Validate(_ string) ([]string, error) {
+	// check that all creds have both usernames and passwords set. Note a credential can have both fields empty
+	for _, cred := range cfg.Credentials {
+		if cred.Pass != "" && cred.User == "" {
+			return nil, fmt.Errorf("credential missing username, has password %v", cred.Pass)
```

Comment:
Did mobile/customer say that username only is an acceptable input?

Reply:
rand mentioned it was acceptable

Reply:
ok I've never tried our discovery flow with just a username. Curious at what the result and output off this branch would be

Reply:
tested with the service, though I don't have any cameras with only a username the `DiscoverCameras` can  still return the insecure cameras when configured this way

Reply:
gotcha

as long as the returned rtsp addresses are usable that's good